thefuck/rules/git_push.py

```diff
@@ -6,7 +6,7 @@
 @git_support
 def match(command):
     return ('push' in command.script
-            and 'set-upstream' in command.output)
+            and 'git push --set-upstream' in command.output)
```

Comment:
I'm not sure how this change relates to the tests that were added. If I undo the changes on this line, the new tests still pass. Can you help me understand how this fixes the issue?

Reply:
I added a test to see if we would match on the bitbucket output, which suggests we won't - but thought it best to leave it.  I can only assume that the output provided was not the entire output, but the only text checked for was `set-upstream`.

Change #682 modified git_push.py to regex search for `git push`, assuming it was there.  You can't do that when the only thing we ever check for in the output is `set-upstream`.

Reply:
I guess a much better example might be if the branch was called `feature/set-upstream`.